File: 201-300/300-longest-increasing-subsequence.cpp
int lengthOfLIS(vector<int>& nums) {
    int n = nums.size();
    if (n == 1) return 1;
    vector<int> dp(n, 1);
    int maxLength = 0;
    for (int i = 1; i < n; ++i) {
        for (int j = 0; j < i; ++j) {
            if (nums[j] < nums[i]) {
                dp[i] = max(dp[j] + 1, dp[i]);
            }
        }
        maxLength = max(maxLength, dp[i]);
    }

    return maxLength;
}

File: python version/300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        max_length = 1
        n = len(nums)
        dp = [1]*n
        for i in range(n):
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j]+1)
            
            max_length = max(max_length, dp[i])

        return max_length